network/common/src/test/java/org/apache/spark/network/TransportClientFactorySuite.java

```diff
@@ -177,4 +177,14 @@ public void closeBlockClientsWithFactory() throws IOException {
     assertFalse(c1.isActive());
     assertFalse(c2.isActive());
   }
+
+  @Test
+  public void closeIdleConnectionForRequestTimeOut() throws IOException, InterruptedException {
+    TransportClientFactory factory = context.createClientFactory();
+    TransportClient c1 = factory.createClient(TestUtils.getLocalHost(), server1.getPort());
+    assertTrue(c1.isActive());
+    Thread.sleep(conf.connectionTimeoutMs());
+    assertFalse(c1.isActive());
```

Comment:
This is flakey. You should check it with a timeout and also use a short connection timeout to make this test fast, such as
```Java
  @Test
  public void closeIdleConnectionForRequestTimeOut() throws IOException, InterruptedException {
    TransportConf conf = new TransportConf(new ConfigProvider() {

      @Override
      public String get(String name) {
        if ("spark.shuffle.io.connectionTimeout".equals(name)) {
          // We should make sure there is enough time for us to observe the channel is active
          return "1s";
        }
        String value = System.getProperty(name);
        if (value == null) {
          throw new NoSuchElementException(name);
        }
        return value;
      }
    });
    TransportContext context = new TransportContext(conf, new NoOpRpcHandler());
    TransportClientFactory factory = context.createClientFactory();
    try {
      TransportClient c1 = factory.createClient(TestUtils.getLocalHost(), server1.getPort());
      assertTrue(c1.isActive());
      long expiredTime = System.currentTimeMillis() + 10000; // 10 seconds
      while (c1.isActive() && System.currentTimeMillis() < expiredTime) {
        Thread.sleep(10);
      }
      assertFalse(c1.isActive());
    } finally {
      factory.close();
    }
  }
```